src/setup/rgw.py

```diff
@@ -14,77 +14,220 @@
+                Log.info(f'rgw_lock value - {rgw_lock_val}')
+                if rgw_lock_val is None:
+                    Log.info(f'Setting confstore value for key :{rgw_lock_key}'
+                        f' and value as :{Rgw._machine_id}')
+                    Rgw._update_config(
+                        rgw_consul_idx, consul_url, [(rgw_lock_key, Rgw._machine_id)])
+                    Log.info('Updated confstore with latest value')
+                    time.sleep(3)
+                    continue
+                elif rgw_lock_val == Rgw._machine_id:
+                    Log.info('Found lock acquired successfully hence processing'
+                        ' with RGW admin user creation.')
+                    rgw_lock = True
+                    break
+                elif rgw_lock_val != Rgw._machine_id:
+                    Log.info('Skipping rgw user creation, as rgw lock is already'
+                        ' acquired by {rgw_lock_val}')
+                    rgw_lock = False
+                    break
+
+            except Exception as e:
+                Log.error('Exception occured while connecting to consul service'
+                    f' endpoint {e}')
+                break
+        if rgw_lock is True:
+            Log.info('Creating admin user.')
+            # TODO: Add rgw admin user creation.
+            Log.info('User is created.')
+            Log.info(f'Deleting rgw_lock key {rgw_lock_key}.')
+            Conf.delete(rgw_consul_idx, rgw_lock_key)
+            Log.info(f'{rgw_lock_key} key is deleted')
+
+        Log.info('Init phase completed.')
         return 0
 
     @staticmethod
-    def test(config_path: str, plan: str):
-        """
-        Perform configuration testing.
-        """
+    def test(cortx_gconf: MappedConf, plan: str):
+        """Perform configuration testing."""
 
+        Log.info('Test phase completed.')
         return 0
 
     @staticmethod
-    def reset(config_path: str):
-        """
-        Remove/Delete all the data/logs that was created by user/testing.
-        """
+    def reset(cortx_gconf: MappedConf):
+        """Remove/Delete all the data/logs that was created by user/testing."""
 
+        Log.info('Reset phase completed.')
         return 0
 
     @staticmethod
-    def cleanup(pre_factory: bool, config_path: str):
-        """
-        Remove/Delete all the data that was created after post install.
-        """
+    def cleanup(cortx_gconf: MappedConf, pre_factory: bool):
+        """Remove/Delete all the data that was created after post install."""
 
+        Log.info('Cleanup phase completed.')
         return 0
 
     @staticmethod
-    def upgrade(config_path: str):
-        """
-        Perform upgrade steps.
-        """
+    def upgrade(cortx_gconf: MappedConf):
+        """Perform upgrade steps."""
 
+        Log.info('Upgrade phase completed.')
         return 0
+
+    @staticmethod
+    def _get_consul_url(cortx_gconf: MappedConf):
```

Comment:
Remove cortx from the mention. Simplify the logic

```
def _get_consul_url(conf: MappedConf, seq: int=0):
"""Return consul url."""
endpoints = conf.get('cortx>external>consul>endpoints')
http_endpoints = list(filter(lambda x: urlparse(x).scheme == 'http', endpoints))
if len(http_endpoints) == 0: 
  raise SetupError(.."consul http endpoint is not specified in the conf. listed endpoints: {endpoints}")
return http_endpoints[seq]
```

Reply:
Done.